server/routes/contacts/create/enter-dob/createContactEnterDobSchemas.ts

```diff
@@ -2,39 +2,61 @@ import z from 'zod'
 import { createSchema } from '../../../../middleware/validationMiddleware'
 
 const SELECT_IS_DOB_KNOWN_MESSAGE = 'Select whether the date of birth is known'
-const DAY_TYPE_MESSAGE = 'Enter a valid day of the month'
-const MONTH_TYPE_MESSAGE = 'Enter a valid month'
-const YEAR_TYPE_MESSAGE = 'Enter a valid year'
-const DAY_REQUIRED_MESSAGE = 'Enter day'
-const MONTH_REQUIRED_MESSAGE = 'Enter month'
-const YEAR_REQUIRED_MESSAGE = 'Enter year'
+const DAY_TYPE_MESSAGE = 'Enter a valid day of the month (1-31)'
+const MONTH_TYPE_MESSAGE = 'Enter a valid month (1-12)'
+const YEAR_TYPE_MESSAGE = 'Enter a valid year. Must be at least 1900'
+const DOB_IN_FUTURE_MESSAGE = 'The date of birth must not be in the future'
 
 export const createContactEnterDobSchema = () => async () => {
   return createSchema({
     isDobKnown: z.string({ message: SELECT_IS_DOB_KNOWN_MESSAGE }).transform(value => value === 'true'),
-    day: z.coerce.number({ message: DAY_TYPE_MESSAGE }).optional(),
-    month: z.coerce.number({ message: MONTH_TYPE_MESSAGE }).optional(),
-    year: z.coerce.number({ message: YEAR_TYPE_MESSAGE }).optional(),
+    day: z.union(
+      [
+        z.literal(''),
+        z.coerce.number({ message: DAY_TYPE_MESSAGE }).min(1, DAY_TYPE_MESSAGE).max(31, DAY_TYPE_MESSAGE).optional(),
```

Comment:
formatting to make it into new lines
`.coerce.number({ message: DAY_TYPE_MESSAGE })

.min(1, DAY_TYPE_MESSAGE)

.max(31, DAY_TYPE_MESSAGE)

.optional(),`

Reply:
I did but then the formatter put it back like this 🤷